sdk-tests/src/test/java/io/dapr/it/testcontainers/workflows/crossapp/WorkflowsCrossAppCallActivityIT.java

```diff
@@ -0,0 +1,278 @@
+    
+    // Start all worker containers
+    crossappWorker.start();
+    app2Worker.start();
+    app3Worker.start();
+  }
+
+  @AfterAll
+  public static void tearDown() {
+    if (crossappWorker != null) {
+      crossappWorker.stop();
+    }
+    if (app2Worker != null) {
+      app2Worker.stop();
+    }
+    if (app3Worker != null) {
+      app3Worker.stop();
+    }
+    if (MAIN_WORKFLOW_CONTAINER != null) {
+      MAIN_WORKFLOW_CONTAINER.stop();
+    }
+    if (APP2_CONTAINER != null) {
+      APP2_CONTAINER.stop();
+    }
+    if (APP3_CONTAINER != null) {
+      APP3_CONTAINER.stop();
+    }
+    if (sharedPlacementContainer != null) {
+      sharedPlacementContainer.stop();
+    }
+    if (sharedSchedulerContainer != null) {
+      sharedSchedulerContainer.stop();
+    }
+  }
+
+  @Test
+  public void testCrossAppWorkflow() throws Exception {
+    // TestContainers wait strategies ensure all containers are ready before this test runs
+
+    String input = "Hello World";
+    String expectedOutput = "HELLO WORLD [TRANSFORMED BY APP2] [FINALIZED BY APP3]";
+    
+    // Create workflow client connected to the main workflow orchestrator
+    // Use the same endpoint configuration that the workers use
+    // The workers use host.testcontainers.internal:50001, so we need to use the mapped port
+    String grpcEndpoint = "localhost:" + MAIN_WORKFLOW_CONTAINER.getMappedPort(50001);
+    String httpEndpoint = "localhost:" + MAIN_WORKFLOW_CONTAINER.getMappedPort(3500);
+    System.setProperty("dapr.grpc.endpoint", grpcEndpoint);
+    System.setProperty("dapr.http.endpoint", httpEndpoint);
+    Map<String, String> propertyOverrides = Map.of(
+        "dapr.grpc.endpoint", grpcEndpoint,
+        "dapr.http.endpoint", httpEndpoint
+    );
+    
+    Properties clientProperties = new Properties(propertyOverrides);
+    DaprWorkflowClient workflowClient = new DaprWorkflowClient(clientProperties);
+    
+    try {
+      String instanceId = workflowClient.scheduleNewWorkflow(CrossAppWorkflow.class, input);
+      assertNotNull(instanceId, "Workflow instance ID should not be null");
+      workflowClient.waitForInstanceStart(instanceId, Duration.ofSeconds(30), false);
+
+      WorkflowInstanceStatus workflowStatus = workflowClient.waitForInstanceCompletion(instanceId, null, true);
+      assertNotNull(workflowStatus, "Workflow status should not be null");
+      assertEquals(WorkflowRuntimeStatus.COMPLETED, workflowStatus.getRuntimeStatus(),
+          "Workflow should complete successfully");
+      String workflowOutput = workflowStatus.readOutputAs(String.class);
+      assertEquals(expectedOutput, workflowOutput, "Workflow output should match expected result");
+    } finally {
+      workflowClient.close();
+    }
+  }
+  
+  /**
+   * Ensures that dependencies are copied to target/dependency for container classpath.
+   * This is needed because the containers need access to the workflow runtime classes.
+   */
```

Comment:
I am wondering why is this required? Can't we make sure that all the dependencies are in the right place during build, so the container just copies the files without having to invoke Maven behind the scenes.